source/gui/guiHelper.py

```diff
@@ -363,7 +362,24 @@ def addDialogDismissButtons(self, buttons, separated=False):
 		self.dialogDismissButtonsAdded = True
 		return buttons
 
+
+def safeAppExit():

```

Comment:
So overall, this ensures that the main loop is stopped, and the top window is destroyed. Could you please add a doctring here.

Does the order of mainloop stop vs window destroyed matter?

Reply:
Added this comment. 

The order does matter. I believe the problem is `safeAppExit` is called from the destruction of several frames which may be the top window, so if the top window is currently being destroyed the correct process is to finally exit the main loop.